doc/api/index.rst

```diff
@@ -225,6 +225,7 @@ and store them in GMT's user data directory.
     datasets.load_earth_geoid
     datasets.load_earth_magnetic_anomaly
     datasets.load_earth_relief
+    datasets.load_earth_wdmam
```

Comment:
Similar to the comment https://github.com/GenericMappingTools/pygmt/pull/2240/files#r1047080605, the name `load_earth_wdmam` is not clear, but `load_earth_world_digital_magnetic_anomaly_map` is too long

Reply:
I figure that if it's the name for the GMT dataset, it shouldn't be too difficult to use that for the PyGMT function to download it. I think I would feel more concerned if it was commonly used data set like `load_earth_relief`, but I believe that users looking for the WDMAM dataset will understand the meaning behind `load_earth_wdmam`.

Reply:
@GenericMappingTools/pygmt-maintainers Any ideas on a good function name?

Reply:
Hm, currently not really. The name `load_earth_world_digital_magnetic_anomaly_map` is quite long. However, the shortcut `wdmam` is not or only for "experienced" users understandable. `load_earth_magnetic_anomaly` is used for [EMAG2 Global Earth Magnetic Anomaly Model](https://www.pygmt.org/dev/api/generated/pygmt.datasets.load_earth_magnetic_anomaly.html#pygmt.datasets.load_earth_magnetic_anomaly), in which `model` was skipped. Do we need both `earth` and `world`? So maybe `load_word_digital_magnetic_anomaly`? This is not really short, but at least a bit more in the length range of [load_earth_vertical_gravity_gradient](https://www.pygmt.org/dev/api/generated/pygmt.datasets.load_earth_vertical_gravity_gradient.html#pygmt.datasets.load_earth_vertical_gravity_gradient).

Reply:
I think `load_world_digital_magnetic_anomaly` could be easily confused with `load_earth_magnetic_anomaly`.

Reply:
I know very little about these two datasets, but from the papers (https://doi.org/10.1186/s40623-016-0404-6, https://doi.org/10.1029/2009GC002471):

> Note that EMAG2 represents magnetic anomalies at 4 km altitude above the geoid in 2–arc min resolution, while the WDMAM grid was specified at 5 km altitude in 3–arc min resolution.

> Conversely, the Earth Magnetic Anomaly Map EMAG-2 ([Maus et al. 2009](https://earth-planets-space.springeropen.com/articles/10.1186/s40623-016-0404-6#ref-CR11)) interpolates between sparse marine magnetic data using directional gridding and extrapolation based on the age map of the seafloor ([Müller et al. 2008](https://earth-planets-space.springeropen.com/articles/10.1186/s40623-016-0404-6#ref-CR14)), but the result displays artificially elongated features due to the improper interpolation of anomalies unrelated to seafloor spreading (associated, for instance, to seamounts).

As I understand them, both EMAG-2 and WDMAM are magnetic anomaly models, but the details diff. So, just like `load_earth_relief`, perhaps we should add the `data_source` parameter (valid values are `emag2` and `wdmam`) to the `load_earth_magnetic_anomaly` function.

Reply:
That's a good idea.